### ISSUE TITLE
Address issues found by SpotBugs

### ISSUE THREAD

**Issue by tomwhite** [samtools/htsjdk#1267]
I ran SpotBugs on the htsjdk source and it found a number of issues that should be looked at.

To gnerate the files, checkout https://github.com/tomwhite/htsjdk/commit/8d2f7255ca062fded660676c5fa6c6635c80da08, and type 

```
./gradlew check -x test
```

Here are the reports that it generated (converted to PDF since GitHub doesn't support HTML attachments; however this means you can't drill down into each issue):
[main.pdf](https://github.com/samtools/htsjdk/files/2802757/main.pdf)
[test.pdf](https://github.com/samtools/htsjdk/files/2802758/test.pdf)

**Comment by cmnbroad** [comment]
Thanks Tom. @jmthibault79 - Might be worth taking a look at the CRAM package part of this as you're refactoring.

**Comment by tomwhite** [comment]
With the latest PRs all the spotbugs issues in the tests have been resolved, and there are now 19 remaining issues in the main code. I've opened issues for the CRAM ones, but there are a few more (with less obvious fixes) that need looking at - they can either be fixed or an exclusion can be added to gradle/spotbugs-exclude.xml. @lbergelson it would be good to get your opinion on these. (The easiest way to get the latest report is to checkout https://github.com/tomwhite/htsjdk/tree/spotbugs and run `./gradlew spotbugsMain`.)

When all the issues have been fixed then we can merge https://github.com/tomwhite/htsjdk/tree/spotbugs to run spotbugs on each new PR to ensure no new issues are introduced.
